pycode/memilio-epidata/memilio/epidata/getDataIntoPandasDataFrame.py

```diff
@@ -202,11 +202,16 @@ def download_file(
                 "Error: URL " + url + " could not be opened.")
     if req.status_code != 200:  # e.g. 404
         raise requests.exceptions.HTTPError("HTTPError: "+str(req.status_code))
-    # get file size from http header
-    # this is only the number of bytes downloaded, the size of the actual file
-    # may be larger (e.g. when 'content-encoding' is gzip; decoding is handled
-    # by iter_content)
-    file_size = int(req.headers.get('content-length'))
+    if 'content-length' in req.headers:
+        # get file size from http header
+        # this is only the number of bytes downloaded, the size of the actual file
+        # may be larger (e.g. when 'content-encoding' is gzip; decoding is handled
+        # by iter_content)
+        file_size = int(req.headers.get('content-length'))
+        # if content length is not known, a progress cant be set.
+        set_progr = True
+    else:
+        set_progr = False
     file = bytearray()  # file to be downloaded
     if progress_function:
```

Comment:
```suggestion
        file_size = int(req.headers.get('content-length'))
    else:
        # if content length is not known, a progress cant be set.
        progress_function(0.0)
        progress_function = None
    file = bytearray()  # file to be downloaded
    if progress_function:
```
so we don't need the `if set_progr` in line 223?
Alternatively
```suggestion
        file_size = int(req.headers.get('content-length'))
        set_progr = True
    else:
        # if content length is not known, a progress cant be set.
        set_progr = False
        if progress_function:
            progress_function(0.0)
    file = bytearray()  # file to be downloaded
    if progress_function and set_progr:
```
I am not sure whether we should/have to set progress_function(0.0), maybe this can be omitted.

Reply:
I dont think we should overwrite the progress function here. 
But merging this with the not progress tracked download is better.